metrics/sink_test.go

```diff
@@ -203,6 +176,61 @@ func TestTrendSink(t *testing.T) {
 	})
 }
 
+func BenchmarkTrendSink(b *testing.B) {
+	// Produce a fixed random values set for the benchmark.
+	values := make([]float64, 0, 10001)
+	for i := 0; i < 10000; i++ {
+		values = append(values, rand.Float64()*100)
+	}
+
+	b.Run("P", func(b *testing.B) {
+		// Prepare a sample size table to control the
+		// variation of the number of samples for each
+		// benchmark.
+		var table = []struct {
+			percentile float64
+			sinkSize   int
+		}{
+			// 50th percentile
+			// Note that computing the 50th percentile is equivalent
+			// to computing the median.
+			{percentile: 0.5, sinkSize: 10},
+			{percentile: 0.5, sinkSize: 100},
+			{percentile: 0.5, sinkSize: 1000},
+			{percentile: 0.5, sinkSize: 10000},
+
+			// 90th percentile
+			{percentile: 0.9, sinkSize: 10},
+			{percentile: 0.9, sinkSize: 100},
+			{percentile: 0.9, sinkSize: 1000},
+			{percentile: 0.9, sinkSize: 10000},
+
+			// 95th percentile
+			{percentile: 0.95, sinkSize: 10},
+			{percentile: 0.95, sinkSize: 100},
+			{percentile: 0.95, sinkSize: 1000},
+			{percentile: 0.95, sinkSize: 10000},
+
+			// 99th percentile
+			{percentile: 0.99, sinkSize: 10},
+			{percentile: 0.99, sinkSize: 100},
+			{percentile: 0.99, sinkSize: 1000},
+			{percentile: 0.99, sinkSize: 10000},
+		}
+
+		for _, v := range table {
+			b.Run(fmt.Sprintf("%d_with_%d_values", int(v.percentile*100), v.sinkSize), func(b *testing.B) {
+				sink := TrendSink{Values: values[0:v.sinkSize]}
+
+				b.ResetTimer()
+				for i := 0; i < b.N; i++ {
+					sink.P(v.percentile)
+				}
+			})
+		}
+	})
+}
```

Comment:
Not certain what this benchmarks as after the first `sink.P` the sink will be "sorted" (after my other proposed change) and as such it will be super fast  .... 

This are also quite a big number of cases in the table that will make running the benchmarks take quite a while, and I am not certain why we would want to do it for this particular inputs.

Also Trend sizes of < 10k are likely not interesting at all. 

I am for this benchmark basically being dropped as I see it as just taking more time in running the benchmarks. And I can tell you from experience that it already takes too long if you want to have a `-count 5` for any statistical significance.

Arguably we should even go and clean some of the other ones that *likely* aren't all that interesting before or with #2751

Reply:
Fair enough, I kept it as a relic of illustrating the performance difference between `Calc` and `P`, but you're right, it doesn't add much value outside of this context 👍🏻